upstairs/src/test.rs

```diff
@@ -907,7 +909,16 @@ mod up_test {
             vec![],
             10,
             vec![request.clone()],
-            ImpactedBlocks::default(),
+            ImpactedBlocks::new(
```

Comment:
small nit: I see this same pattern repeated quite a few times.
Is there any way to abstract it out to a single place that everyone uses?

Reply:
So it's worth noting that the previous use of `::default()` was just incorrect. it was issuing these commands and stating that they impacted zero blocks, which was just wrong.

Looking back on it I actually found another instance where I replaced a default() with an Empty incorrectly.

Maybe we should have a function that takes a ReadRequest or WriteRequest and returns the ImpactedBlocks that covers it

This wouldn't be particularly helpful to have outside of the test code, because those requests are always one block long. Creating a bunch of ImpactedBlocks from that in release code would be horribly inefficient. But in the test code which is all doing one-block things anyway, it would help a bit.

This raises some questions about how create_read_eob and create_write_eob are called in general in my mind, seeing how easy it is to accidentally write code that passes in uncorrelated requests and impactedblocks into it. But i think that further question is outside the scope of this PR.

Reply:
You bring up good points here, but I was thinking something simple like this:

```
    fn generic_impacted_blocks() -> ImpactedBlocks {
        ImpactedBlocks::new(
            ImpactedAddr { 
                extent_id: 0,
                block: 7,
            },  
            ImpactedAddr { 
                extent_id: 0,
                block: 7,
            },
        )
    }
```

Then call `create_read_eob()` like this:
```
        let op = create_read_eob(
            next_id,
            vec![],
            10,
            vec![request.clone()],
            generic_impacted_blocks(),
        );  
```

Just as a way of having a little more code density.
It was a "small nit", so not something I was really concerned about :)